networkx/generators/classic.py

```diff
@@ -566,6 +620,91 @@ def empty_graph(n=0, create_using=None, default=Graph):
     return G
 
 
+@nx._dispatch(graphs=None)
+def full_rary_tree(r, n, create_using=None):
+    """Creates a full r-ary tree of `n` nodes.
+
+    Sometimes called a k-ary, n-ary, or m-ary tree.
+    "... all non-leaf nodes have exactly r children and all levels
+    are full except for some rightmost position of the bottom level
+    (if a leaf at the bottom level is missing, then so are all of the
+    leaves to its right." [1]_
+
+    Parameters
+    ----------
+    r : int
+        branching factor of the tree
+    n : int
+        Number of nodes in the tree
+    create_using : NetworkX graph constructor, optional (default=nx.Graph)
+       Graph type to create. If graph instance, then cleared before populated.
+
+    Returns
+    -------
+    G : networkx Graph
+        An r-ary tree with n nodes
+
+    References
+    ----------
+    .. [1] An introduction to data structures and algorithms,
+           James Andrew Storer,  Birkhauser Boston 2001, (page 225).
+    """
+    G = empty_graph(n, create_using)
+    G.add_edges_from(_tree_edges(n, r))
+    return G
+
+
+def kneser_graph(n, k):
+    """Returns the Kneser Graph with parameters $n$ and $k$.
+
+    The Kneser Graph is the graph whose vertices are the $k$-subsets of
+    range($n$), such that two vertices are adjacent if their corresponding sets
+    are disjoint.
```

Comment:
We use the word `node` over `vertex`.
It might be good to define `$k$-subsets` and we usually don't use latex for single letters.
Ironically, I'd prefer `{0, 1, 2, ..., n-1}` to `range(n)` in the docs... Or maybe in words instead.
Perhaps something like:
```
    The Kneser Graph has nodes that are k-tuples (subsets) of the integers
    between 0 and n-1. Nodes are adjacent if their corresponding sets are disjoint.
```

Reply:
done, thanks for the suggestion. I agree that using ``range(n)`` isn't a good idea, especially in the docs.